src/hotspot/os_cpu/linux_aarch64/atomic_linux_aarch64.hpp

```diff
@@ -27,57 +27,184 @@
 #define OS_CPU_LINUX_AARCH64_ATOMIC_LINUX_AARCH64_HPP
 
 #include "runtime/vm_version.hpp"
+#include "atomic_aarch64.hpp"
 
 // Implementation of class atomic
+
+// Atomic stub implementation.
+// Default implementations are in atomic_aarch64.S
+//
+// All stubs pass arguments the same way
+// x0: src/dest address
+// x1: arg1
+// x2: arg2 (optional)
+// x3, x8, x9: scratch
+typedef uint64_t (*aarch64_atomic_stub_t)(volatile void *ptr, uint64_t arg1, uint64_t arg2);
+
+// Pointers to stubs
+extern aarch64_atomic_stub_t aarch64_atomic_fetch_add_4_impl;
+extern aarch64_atomic_stub_t aarch64_atomic_fetch_add_8_impl;
+extern aarch64_atomic_stub_t aarch64_atomic_xchg_4_impl;
+extern aarch64_atomic_stub_t aarch64_atomic_xchg_8_impl;
+extern aarch64_atomic_stub_t aarch64_atomic_cmpxchg_1_impl;
+extern aarch64_atomic_stub_t aarch64_atomic_cmpxchg_4_impl;
+extern aarch64_atomic_stub_t aarch64_atomic_cmpxchg_8_impl;
+
```

Comment:
I don't think you need to duplicate all these declarations here if you include `"atomic_aarch64.hpp"` which already declares all these types and variables.

Reply:
Thanks, done.